src/components/invoices/ExportInvoicesDialog.tsx

```diff
@@ -0,0 +1,216 @@
+import { gql } from '@apollo/client'
+import { useFormik } from 'formik'
+import { forwardRef } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import styled from 'styled-components'
+import { object, string } from 'yup'
+
+import { Button, Dialog, Typography } from '~/components/designSystem'
+import { addToast } from '~/core/apolloClient'
+import {
+  CreateDataExportsInvoicesInput,
+  DataExportFormatTypeEnum,
+  ExportTypeEnum,
+  useCreateInvoicesDataExportMutation,
+} from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { useCurrentUser } from '~/hooks/useCurrentUser'
+import { theme } from '~/styles'
+
+import { formatFiltersForInvoiceQuery } from '../designSystem/Filters/utils'
+import { RadioGroupField } from '../form'
+
+gql`
+  mutation createInvoicesDataExport($input: CreateDataExportsInvoicesInput!) {
+    createInvoicesDataExport(input: $input) {
+      id
+    }
+  }
+`
+
+type ExportInvoicesDialogProps = {
+  invoicesTotalCount: number | undefined
+  invoicesVariablesSearchTerm: string | null | undefined
+}
+
+export interface ExportInvoicesDialogRef {
+  openDialog: () => unknown
+  closeDialog: () => unknown
+}
+
+export const ExportInvoicesDialog = forwardRef<ExportInvoicesDialogRef, ExportInvoicesDialogProps>(
+  ({ invoicesTotalCount, invoicesVariablesSearchTerm }: ExportInvoicesDialogProps, ref) => {
+    const { translate } = useInternationalization()
+    const { currentUser } = useCurrentUser()
+    let [searchParams] = useSearchParams()
+    const [triggerCreateInvoicesDataExport] = useCreateInvoicesDataExportMutation({
+      onCompleted({ createInvoicesDataExport }) {
+        if (createInvoicesDataExport) {
+          addToast({
+            message: translate('text_66b323b63e76c400f78cd342'),
+            severity: 'info',
+          })
+        }
+      },
+    })
+
+    const formikProps = useFormik<Omit<CreateDataExportsInvoicesInput, 'filters'>>({
+      initialValues: {
+        format: DataExportFormatTypeEnum.Csv,
+        resourceType: ExportTypeEnum.Invoices,
+      },
+      validationSchema: object().shape({
+        format: string().required(''),
+        resourceType: string().required(''),
```

Comment:
I think you can do something like this to be more restrictive: 

```suggestion
        resourceType: string().valid(...Object.values(ExportTypeEnum)).required('')
```

Reply:
`resourceType: string().oneOf(Object.values(ExportTypeEnum)).required(''),` did the trick.

Good idea for filters too, I'll change separately thanks!